packages/mui-material/src/Tooltip/Tooltip.js

```diff
@@ -631,7 +632,11 @@ const Tooltip = React.forwardRef(function Tooltip(inProps, ref) {
 
   const popperProps = appendOwnerState(
     PopperComponent,
-    { ...PopperProps, ...componentsProps.popper },
+    {
+      ...PopperProps,
+      ...componentsProps.popper,
+      sx: { ...sx, ...PopperProps.sx, ...componentsProps.popper?.sx },
```

Comment:
We should probably do a check for `isHostComponent` too, we don't want to propagate this on native elements. cc @michaldudak we probably need a more general purpose append utility, that would work with props other than `ownerState` :)

In this case I believe we need to change this too:
```suggestion
      ...(isHostComponent(PopperComponent) && {
        sx: { ...sx, ...PopperProps.sx, ...componentsProps.popper?.sx },
      }),  
```

We can import the `isHostComponent` from `@mui/core`:
```
import { isHostComponent } from '@mui/core';
```

Reply:
@michaldudak we should have merge here the `style` prop.

Reply:
Right, a utility that could merge `style`, `sx`, `className`, `ownerProps` and (possibly) event handlers from two props objects could indeed be useful in this case.